### PR TITLE
Added some more documentation to the linear_util module

### PR DESCRIPTION
**Pull request by gnecula**
Also cleaned up the inconsistent way of importing the module.
Prefer importing with qualified name 'lu.transformation' rather
than just 'transformation'.

I found the need for such documentation while trying to understand how WrappedFun caching works for #1945.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
LGTM, the docs are a big improvement.

**Comment by mattjj** [comment]
It might be worth also waiting for a review from @dougalm.

File: tests/util_test.py
Comment (mattjj) [inline tests/util_test.py]: It's 2020 so we could just drop py2 compatibility, but to make this work in py2 I think we need something like

```python
def f(*args, **kwargs):
  factor = kwargs.pop('factor', 2)
  assert not kwargs
  # ...
```

**Review by dougalm — APPROVED** [review]
Thanks George! This looks great to me.

**Pull request merged**
